packages/core/src/modules/vc/libraries/documentLoader.ts

```diff
@@ -1,8 +1,50 @@
+import type { AgentContext } from '../../../agent/context/AgentContext'
 import type { DocumentLoader } from './jsonld'
 
-export function getDocumentLoader(): () => DocumentLoader {
-  // eslint-disable-next-line @typescript-eslint/no-var-requires
-  const loader = require('@digitalcredentials/jsonld/lib/documentLoaders/node')
+import { AriesFrameworkError } from '../../../error/AriesFrameworkError'
+import { DidResolverService } from '../../dids'
 
-  return loader as () => DocumentLoader
+import jsonld from './jsonld'
+import { getNativeDocumentLoader } from './nativeDocumentLoader'
+
+export type DocumentLoaderWithContext = (agentContext: AgentContext) => DocumentLoader
+
+export function defaultDocumentLoader(agentContext: AgentContext): DocumentLoader {
```

Comment:
Wouldn't it make more sense to create a class here? I am not sure about the this scope and if that would case issues, but it might make it cleaner here. If it causes issues this is also okay ofcourse.

Reply:
Well the problem is that you need to return a document loader function. The jsonld library only provides the url, so we basically have a wrapper around the jsonld method. Making it a class would be a bit weird as you would have a class with one public method that returns another method.

I'll revisit this soon